src/main/java/org/madhuri/app/message/handler/StockHandler.java

```diff
@@ -31,5 +37,29 @@ public Response getStockInfo(@PathParam("symbol") String symbol) {
             return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
         }
     }
+    
+    @GET
+    @Path("/search")
+    public Response searchStocks(@QueryParam("q") String query) {
+        try {
+            if (query == null || query.trim().isEmpty()) {
+                return Response.status(Response.Status.BAD_REQUEST)
+                               .entity("Query parameter 'q' cannot be empty.")
+                               .build();
+            }
+
+            StockResponse stockResponse = stockService.searchSymbols(query);
+
+            if (stockResponse != null && stockResponse.getResult() != null && !stockResponse.getResult().isEmpty()) {
```

Comment:
Why are you doing getResults twice here ? Can we not assign a variable and nil check and use same for assignment ?